src/detect/ScanI2CTwoWire.cpp

```diff
@@ -292,7 +292,18 @@ void ScanI2CTwoWire::scanPort(I2CPort port)
 
                 break;
 
-                SCAN_SIMPLE_CASE(SHT31_ADDR, SHT31, "SHT31 sensor found\n")
+            case SHT31_4x_ADDR:
+                registerValue = getRegisterValue(ScanI2CTwoWire::RegisterLocation(addr, 0x89), 2);
+                if (registerValue == 0x11a2) {
+                    type = SHT4X;
+                    LOG_INFO("SHT4X sensor found\n");
+                } else {
+                    type = SHT31;
+                    LOG_INFO("SHT31 accelerometer found\n");
+                }
+
+                break;
+
```

Comment:
If this only works for the SHT45 sensor as per your comment, please name the macro accordingly to leave room for 41 and 42

Reply:
SHT31 and all SHT4X share the same ADDR (0x44).
Unfortunately I don't have the other sensors to test for the registerValue but like all SHT4x sensors use the same library, i would go for something like:
`if (registerValue == 0x11a2 || registerValue == <sht40> || registerValue == <sht41>)`